Content.Client/Revolutionary/RevolutionarySystem.cs

```diff
@@ -0,0 +1,29 @@
+using Content.Shared.Revolutionary.Components;
+using Content.Client.Antag;
+using Content.Shared.StatusIcon.Components;
+
+namespace Content.Client.Revolutionary;
+
+/// <summary>
+/// Used for the client to get status icons from other revs.
+/// </summary>
+public sealed class RevolutionarySystem : AntagStatusIcons<RevolutionaryComponent, HeadRevolutionaryComponent>
+{
+    protected override void GetStatusIcon(EntityUid uid, RevolutionaryComponent antag, HeadRevolutionaryComponent? leader, string antagStatusIcon, string? antagLeaderStatusIcon, ref GetStatusIconsEvent args)
+    {
+        base.GetStatusIcon(uid, antag, leader, antagStatusIcon, antagLeaderStatusIcon, ref args);
+    }
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<RevolutionaryComponent, GetStatusIconsEvent>(OnGetStatusIcon);
+    }
+
+    /// <summary>
+    /// Checks if you have the revolutionary or head rev component and gets status icons from other revs. 
+    /// </summary>
+    private void OnGetStatusIcon(EntityUid uid, RevolutionaryComponent component, ref GetStatusIconsEvent args)
+    {
+        GetStatusIcon(uid, component, null, component.RevStatusIcon, component.HeadRevStatusIcon, ref args);
+    }
```

Comment:
This should be handled on the abstract system itself.

Using an interface will also mean you won't need to pass the icon fields in.

Reply:
What do you mean by interface

Reply:
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/interface like a C# interface

Reply:
I don't know man this shit is not clicking in my head gonna make my brain explode.

Reply:
given that its now 2 event handlers for headrev and normal rev, would an interface be needed and if it is, would it be implemented on the rev component headrev component and zombie component?